src/shareManagement.js

```diff
@@ -123,6 +124,49 @@ class Shares {
       })
   }
 
+  /**
+   * Shares a space with specified user
+   * @param   {string}    path             path to the remote file share
+   * @param   {string}    username         name of the user to share with
+   * @param   {string}    spaceId          id of the space
+   * @param   {object}    optionalParams   {permissions: integer, expirationDate: ISO Date, remoteUser: boolean, attributes: assoc array (at free disposal)}
+   * @returns {Promise.<ShareInfo>}        instance of class ShareInfo
+   * @returns {Promise.<error>}            string: error message, if any.
+   */
+  shareSpaceWithUser (path, username, spaceId, optionalParams) {
+    const postData = {
+      shareType: this.helpers.OCS_SHARE_TYPE_SPACE,
+      shareWith: username,
+      space_ref: spaceId
+    }
+
+    if (path) {
+      postData.path = this.helpers._normalizePath(path)
```

Comment:
Do I get it right that a space share is either the entire space or a specific file from the space?

Reply:
Honestly, I haven't dealt with sharing resources within a space yet. I assumed it works the same, so I left the path. I'll gather some information about this

Reply:
As discussed: we keep it as it is for now